blocks/blockoracle.go

```diff
@@ -69,16 +71,24 @@ func (bo *Oracle) BlockEligible(layerID types.LayerID) (types.ATXID, []types.Blo
 	}
 
 	epochNumber := layerID.GetEpoch()
-	bo.log.Info("asked for eligibility for epoch %d (cached: %d)", epochNumber, bo.proofsEpoch)
+	var cachedEpochDescription log.Field
+	if bo.proofsEpoch == DefaultProofsEpoch {
+		cachedEpochDescription = log.String("cached_epoch_id", "(none)")
+	} else {
+		cachedEpochDescription = log.FieldNamed("cached_epoch_id", bo.proofsEpoch.Field())
+	}
```

Comment:
@lrettig I know I already approved this and it's already merged, but I just realized that this approach is problematic.

We sometimes put a string (`"(none)"`) and sometimes a numeric value (`bo.proofsEpoch.Field()` uses `log.Uint64()`) in a field named `cached_epoch_id`.

I think that Elasticsearch doesn't like it when a field has more than one data type and it causes indexing issues. Not sure exactly what the implications are.

@narayanprusty is this really an issue? Is there some better practice for cases like this?

Reply:
I agree we need to be careful about this - I didn't realize this was an issue previously, but it caused problems with another field (an error about late message arrival in hare) in the last testnet. It would be good to know the best practice for cases like this.

Reply:
let's wait for @narayanprusty to chime in

Reply:
If we have different data type for a particular field then ES raises exception and the log is never stored. We have to fix it in code. We can convert uint64 to integer string.

Other solution is to transform the data type using logstash data type transformation plugin. But that isn't a good idea as we have to put install and configure the plugin in CI, spacecraft, local testnet and all other systems that push logs to ELK. In long term it will become difficult to manage this.

Reply:
@lrettig why is it important to have this `"(none)"` string? I know Kibana let's you filter for explicitly missing fields - can't we just omit the field if it's not cached?

Reply:
Opened #2367, let's follow up there